src/python/WMCore/REST/Server.py

```diff
@@ -767,9 +767,14 @@ def _call(self, param):
             raise APINotSpecified()
         api = param.args.pop(0)
         if api not in self.methods[request.method]:
-            response.headers['Allow'] = \
-                " ".join(sorted([m for m, d in self.methods.iteritems() if api in d]))
-            raise APIMethodMismatch()
+            methods = " ".join(sorted([m for m, d in self.methods.iteritems() if api in d]))
+            response.headers['Allow'] = methods
+            if not methods:
+                msg = 'Api "%s" not found. This method supports these Apis: %s' % (api, self.methods[request.method])
```

Comment:
```suggestion
                msg = 'Api "%s" not found. This method supports these Apis: %s' % (api, self.methods[request.method].keys())
```

Reply:
We're not in a hurry to get this in, so I'll let you commit and squash this single change.

Reply:
done